go-controller/pkg/ovn/egressgw.go

```diff
@@ -442,14 +441,34 @@ func (oc *Controller) addPerPodGRSNAT(pod *kapi.Pod, podIfAddrs []*net.IPNet) er
 // by ecmp routes
 func (oc *Controller) addHybridRoutePolicyForPod(podIP net.IP, node string) error {
 	if config.Gateway.Mode == config.GatewayModeLocal {
+		// handle deletion of legacy policies, makes it safe for upgrades
+		err := delLegacyHybridRoutePolicyForPod(podIP, node)
```

Comment:
we should handle this in a sync method: https://github.com/ovn-org/ovn-kubernetes/pull/2348

will update that PR shortly, and get your review on it, then you can build on top of that

Reply:
@trozet : Ack! Will take a look asap.

Reply:
I have reviewed that. I'll rebase as soon as that gets in.